pkg/controllers/machine/disruption/emptiness.go

```diff
@@ -0,0 +1,92 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package disruption
+
+import (
+	"context"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/utils/clock"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	machineutil "github.com/aws/karpenter-core/pkg/utils/machine"
+)
+
+// Emptiness is a machine sub-controller that adds or removes status conditions on empty machines based on TTLSecondsAfterEmpty
+type Emptiness struct {
+	kubeClient client.Client
+	clock      clock.Clock
+}
+
+//nolint:gocyclo
+func (e *Emptiness) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisioner, machine *v1alpha5.Machine) (reconcile.Result, error) {
+	hasEmptyCondition := machine.StatusConditions().GetCondition(v1alpha5.MachineEmpty).IsTrue()
+	// From here there are three scenarios to handle:
+	// 1. If TTLSecondsAfterEmpty is not configured remove the emptiness status condition
+	if provisioner.Spec.TTLSecondsAfterEmpty == nil {
```

Comment:
Just a observation: This is not how it is in the node controller, but I think this is the correct way to do it.